quantum/process_keycode/process_leader.h

```diff
@@ -34,5 +34,6 @@ void qk_leader_start(void);
 
 #define LEADER_EXTERNS() extern bool leading; extern uint16_t leader_time; extern uint16_t leader_sequence[5]; extern uint8_t leader_sequence_size
 #define LEADER_DICTIONARY() if (leading && timer_elapsed(leader_time) > LEADER_TIMEOUT)
+#define LEADER_INFINITE() if (leading && leader_sequence_size > 0 && timer_elapsed(leader_time) > LEADER_TIMEOUT)
```

Comment:
This is still checking the timeout, and not making it "infinite".
Shouldn't it be: 
```suggestion
#define LEADER_INFINITE() if (leading && leader_sequence_size > 0)
```

Reply:
Unless I have missed some magic, I would also expect `LEADER_INFINITE` to be used somewhere.

Reply:
> Unless I have missed some magic, I would also expect `LEADER_INFINITE` to be used somewhere.

@zvecr It should be used on the user space or keymap file. I can not use it directly here. I'm using this on my personal keyboards which are not merged on qmk

> This is still checking the timeout, and not making it "infinite".
> Shouldn't it be:

@drashna it is only infinite for the leader key, not for the entire sequence. I tried to make that clear on the PR description but seems that I failed. The objective is to allow you to hit the leader key, rest as much as you want, and then hit the sequence under the expected timeout. With your suggestion you can reach a state where the sequence will be checked as long as you hit one key.

Reply:
If code is for end users, i would expect the relevant documentation to be updated.

Reply:
@zvecr Yes, I will update the documentation if the technical stuff gets accepted. I'm not saying that we need to merge this PR first, I just mean that I want to be sure that the technical part is OK before writing any docs.

Reply:
And this change request should still happen.  There is no need to check the timer, if you're never actually using or respecting it.

Reply:
> There is no need to check the timer, if you're never actually using or respecting it.

That is not true. I am respecting the timer, but only if the sequence has some content. The timer check is still required because otherwise you will never let the sequence finish, you will enter here as soon as the first key is registered.

In any case, I am thinking about adding some define checks inside the same declaration, and not force the user to use two different macros.